pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -0,0 +1,145 @@
+		inventory: collectors.NewCollectorInventory(),
+		runCfg: &collectors.CollectorRunConfig{
+			APIClient:   chk.apiClient,
+			ClusterID:   chk.clusterID,
+			Config:      chk.orchestratorConfig,
+			MsgGroupRef: &chk.groupID,
+		},
+		stopCh: make(chan struct{}),
+	}
+
+	bundle.prepare()
+
+	return bundle
+}
+
+// prepare initializes the collector bundle internals before it can be used.
+func (cb *CollectorBundle) prepare() {
+	cb.prepareCollectors()
+	cb.prepareExtraSyncTimeout()
+}
+
+// prepareCollectors initializes the bundle collector list.
+func (cb *CollectorBundle) prepareCollectors() {
+	// No collector configured in the check configuration.
+	// Use the list of stable collectors as the default.
+	if len(cb.check.instance.Collectors) == 0 {
+		cb.collectors = cb.inventory.StableCollectors()
+		return
+	}
+
+	// Collectors configured in the check configuration.
+	// Build the custom list of collectors.
+	for _, name := range cb.check.instance.Collectors {
+		if collector, err := cb.inventory.CollectorByName(name); err == nil {
+			if !collector.Metadata().IsStable {
+				_ = cb.check.Warnf("Using unstable collector: %s", name)
+			}
+			cb.collectors = append(cb.collectors, collector)
+		} else {
+			_ = cb.check.Warnf("Unsupported collector: %s", name)
+		}
+	}
+}
+
+// prepareExtraSyncTimeout initializes the bundle extra sync timeout.
+func (cb *CollectorBundle) prepareExtraSyncTimeout() {
+	// No extra timeout set in the check configuration.
+	// Use the default.
+	if cb.check.instance.ExtraSyncTimeoutSeconds <= 0 {
+		cb.extraSyncTimeout = defaultExtraSyncTimeout
+		return
+	}
+
+	// Custom extra timeout.
+	cb.extraSyncTimeout = time.Duration(cb.check.instance.ExtraSyncTimeoutSeconds) * time.Second
+}
+
+// Initialize is used to initialize collectors part of the bundle.
+// During initialization informers are created, started and their cache is
+// synced.
+func (cb *CollectorBundle) Initialize() error {
+	informersToSync := make(map[apiserver.InformerName]cache.SharedInformer)
+
+	for _, collector := range cb.collectors {
+		collector.Init(cb.runCfg)
+		informer := collector.Informer()
+		informersToSync[apiserver.InformerName(collector.Metadata().Name)] = informer
+
+		// we run each enabled informer individually as starting them through the factory
+		// would prevent us to restarting them again if the check is unscheduled/rescheduled
+		// see https://github.com/kubernetes/client-go/blob/3511ef41b1fbe1152ef5cab2c0b950dfd607eea7/informers/factory.go#L64-L66
+		go informer.Run(cb.stopCh)
+	}
+
+	return apiserver.SyncInformers(informersToSync, cb.extraSyncTimeout)
+}
+
+// Run is used to sequentially run all collectors in the bundle.
+func (cb *CollectorBundle) Run(sender aggregator.Sender) {
+	for _, collector := range cb.collectors {
+		runStartTime := time.Now()
+
+		result, err := collector.Run(cb.runCfg)
+		if err != nil {
+			_ = cb.check.Warnf("Collector %s failed to run: %s", collector.Metadata().Name, err.Error())
```

Comment:
I am wondering about the level here. If panics happens, shouldn't we log an error here as well?